knockout-pre-rendered.js

```diff
@@ -121,20 +121,93 @@
     };
   }
 
-  function InitializedForeach(spec) {
+  // Knockout automatically generates "_ko_property_writers" for a subset of its
+  // built-in bindings, registered as "_twoWayBindings". Here, we can compel it to
+  // do the same for the rest of the bindings that we want to support with "init".
+  ko.utils.extend(ko.expressionRewriting._twoWayBindings, {
+    'text': true,
+    'html': true,
+    'visible': true,
+    'enable': true,
+    'disable': true
+  });
+
+  // KO's built-in "_twoWayBindings" logic can only carry us so far. For other bindings, 
+  // we need to generate our own property writers, in much the same way as KO does.
+
+  // Generate property writers for all properties referenced in an "attr" binding.
+  generatePropertyWritersForBinding("attr", "_ko_prerender_attrPropertyWriters", "attr");
+
+  // This method generates a binding preprocessor for the specified binding, and for each
+  // applicable field referenced in the binding params, it generates a writer.
+  function generatePropertyWritersForBinding(bindingName, propertyWritersBindingName, defaultWriterName, mapExpressionsCallback) {
+    // Chain this with any existing preprocessor.
+    var existingPreprocessor = ko.bindingHandlers[bindingName].preprocess;
+
+    ko.bindingHandlers[bindingName].preprocess = function (value, name, addBinding) {
+      if (existingPreprocessor) {
+        value = existingPreprocessor(value, name, addBinding);
+      }
+      var expressions = ko.expressionRewriting.parseObjectLiteral(value);
+      if (mapExpressionsCallback) {
+        expressions = ko.utils.arrayMap(expressions, mapExpressionsCallback);
+      }
+      var writers = [];
+      ko.utils.arrayForEach(expressions, function (expression) {
+        if (expression != null) {
+          var writableExpression = getWritableValue(("unknown" in expression) ? expression["unknown"] : expression.value);
+          if (writableExpression) {
+            writers.push(
+                "'" + (("unknown" in expression) ? defaultWriterName : expression.key) 
```

Comment:
What does this statement do exactly?

Reply:
It has to do with what `ko.utils.parseObjectLiteral` returns for different binding syntax.

For example, if your `init` binding looks like `init: { field: myField, value: 123 }`, then `parseObjectLiteral` will return an array of objects like:

```
[
    {
       key: 'field', 
       value: 'myField' 
    },
   {
       key: 'value', 
       value: '123' 
    },
]
```

But if the binding looks like `init: myField`, then `parseObjectLiteral` returns:
```
[
   { unknown: 'myField' }
]
```
The code on line 161 checks if the object contains a property named `unknown`, and if so, uses the default name passed into the function.  Otherwise, it uses the `.key` property.

Reply:
Ah, thanks for the reply.